### ISSUE TITLE
Modifed LOGIN flow

### ISSUE THREAD

**Issue by JsChiSurf** [MattTW/BlinkMonitorProtocol#39]
Not sure if anyone is still actively monitoring this repository.  Looks like on 5/11/2020 that Blink modified their login process thus disabling to login using the method documented here.  Not sure if anyone has had success in replicating the new flow.

**Comment by axid2006** [comment]
Hello!
sadly i've got the same problem :-(
is it possible to fix this issue - would be very helpful - thx :-)

my error:
Invoke-RestMethod: The remote name could not be resolved: 'prod.immedia-semi.com'
In C: \ Users \ mike\ Desktop \ BlinkVideoDownloader.ps1: 58 characters: 13
+ $ response = Invoke-RestMethod -UseBasicParsing $ uri -Method Post -Hea ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~
     + CategoryInfo: NotSpecified: (:) [Invoke-RestMethod], WebException
     + FullyQualifiedErrorId: System.Net.WebException, Microsoft.PowerShell.Commands.InvokeRestMethodCommand

Invalid credentials provided. Please verify email and password.
Press Enter to continue ...

**Comment by MauiYow** [comment]
I'm getting the same error (US using prod.immedia-semi.com)

Invalid credentials provided. Please verify email and password.
Press Enter to continue ...

**Comment by MauiYow** [comment]
This may help... It took me most of the day to find this solution for the BlinkVideoDownloader.ps1 that was updated to pass a token. Hope this helps...

M

**Comment by MattTW** [comment]
I haven't had a chance to test and update the doc, etc - it looks like the changes are:

- Initial login URL is now:  "https://rest-prod.immedia-semi.com/api/v4/account/login"
- region is passed back in the response in addition to accountID and authToken.   The region must be used in subsequent URLs server endpoint as:  'https://rest-' + region + '.immedia-semi.com'

**Comment by axid2006** [comment]
i've tried it but it doesn't work :-(

i've changed (line 77)

> $uri = 'https://rest-'+ $region +".immedia-semi.com/api/v1/camera/usage" to $uri = 'https://rest-'+ $region +".immedia-semi.com/api/v4/camera/usage"


and (line 124)
```
$uri = 'https://rest-'+ $region +'.immedia-semi.com/api/v1/accounts/'+ $accountID +'/media/changed?since=2015-04-19T23:11:20+0000&page=' + $pageNum to 
$uri = 'https://rest-'+ $region +'.immedia-semi.com/api/v4/accounts/'+ $accountID +'/media/changed?since=2015-04-19T23:11:20+0000&page=' + $pageNum

```
but the same error happens again :-(

-----
Invoke-RestMethod : {"message":"An app update is required"}
In C:\Users\axid\Desktop\BlinkVideoDownloader.ps1:58 Zeichen:13
+ $response = Invoke-RestMethod -UseBasicParsing $uri -Method Post -Hea ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
Invalid credentials provided. Please verify email and password.
Drücken Sie die Eingabetaste, um den Vorgang fortzusetzen...:

**Comment by angezan** [comment]
> I haven't had a chance to test and update the doc, etc - it looks like the changes are:
> 
> * Initial login URL is now:  "https://rest-prod.immedia-semi.com/api/v4/account/login"
> * region is passed back in the response in addition to accountID and authToken.   The region must be used in subsequent URLs server endpoint as:  'https://rest-' + region + '.immedia-semi.com'

I tried it and it works fine: login V4 succesfull 
thanks

**Comment by mordorf** [comment]
Please can someone explain exactly what we need to do to get this working in layman's terms?
Many Thanks :-)

**Comment by MattTW** [comment]
blinkVideoDownload.py has been updated with correct URLs and logic to work with current Blink API.
